### PR TITLE
Inductor cpp wrapper: support LinearBinary

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99957

Support `mkldnn::_linear_pointwise.binary` in cpp wrapper for GPT-J.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]
LGTM. Meanwhile, please check if the code should be moved to mkldnn.py.

**Comment by chunyuan-w** [comment]
> LGTM. Meanwhile, please check if the code should be moved to mkldnn.py.

I checked `mkldnn.py` and it only contains mkldnn related `Module`s while all the mkldnn related `ir`s are still in `ir.py`. cc @XiaobingSuper

**Review by jansel — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

File: torch/_inductor/graph.py [hunk 1]
```diff
@@ -75,8 +75,11 @@ def supported_dtype_of_cpp_wrapper(dtype, cuda):
 
 def may_get_constant_buffer_dtype(constant_buffer):
     assert isinstance(
-        constant_buffer, sympy.Symbol
-    ), "get_constant_buffer_dtype only supports input of sympy.Symbol"
+        constant_buffer, (sympy.Symbol, sympy.core.numbers.Integer)
+    ), "get_constant_buffer_dtype only supports input of sympy.Symbol or sympy.core.numbers.Integer"
+    if isinstance(constant_buffer, sympy.core.numbers.Integer):
+        return torch.int64
+    assert isinstance(constant_buffer, sympy.Symbol)
```
[the remark this answers is not in the document]
Comment (chunyuan-w) [reply torch/_inductor/graph.py]: Removed the redundant assert.

**Pull request closed without merging**